### ISSUE TITLE
refactor(jupyter): deduplicate post-pip jupyter provisioning RUN blocks

### ISSUE THREAD

**Issue by ide-developer** [opendatahub-io/notebooks#3939]
## Summary

Leaf Jupyter workbench Dockerfiles repeat the same **post-`uv pip install` provisioning** heredoc: Elyra/runtime cleanup, kernel `display_name` tweak, `jupyter_server_config.py` install, `jupyter labextension disable`, `labconfig` `chmod`, `apply.sh`, and `ensure-openshift-site-packages.sh`.

The block is copy-pasted across **10** `Dockerfile.konflux.*` leaf images with minor ordering/comment drift. Recent OpenShift permission work ([#3928](https://github.com/opendatahub-io/notebooks/issues/3928) / [#3934](https://github.com/opendatahub-io/notebooks/pull/3934)) had to touch every copy (e.g. `install -m 0664`, `chmod g+w labconfig`, `USER 1001:0`), which is brittle and already caused a regression in **trustyai** where the second `RUN` dropped back to `USER 1001` instead of `USER 1001:0`.

This issue is **not** covered by the existing Dockerfile dedup plan ([#3355](https://github.com/opendatahub-io/notebooks/issues/3355) / [RHOAIENG-54488](https://issues.redhat.com/browse/RHOAIENG-54488)), which targets Konflux twin files and cpu/cuda/rocm **file** merges — not shared **RUN** content inside those files.

## Affected files (today)

All contain a near-identical block after pip install:

- `jupyter/minimal/ubi9-python-3.12/Dockerfile.konflux.{cpu,cuda,rocm}`
- `jupyter/datascience/ubi9-python-3.12/Dockerfile.konflux.cpu`
- `jupyter/trustyai/ubi9-python-3.12/Dockerfile.konflux.cpu` *(split across two `RUN`s — extra footgun)*
- `jupyter/pytorch/ubi9-python-3.12/Dockerfile.konflux.cuda`
- `jupyter/pytorch+llmcompressor/ubi9-python-3.12/Dockerfile.konflux.cuda`
- `jupyter/tensorflow/ubi9-python-3.12/Dockerfile.konflux.cuda`
- `jupyter/rocm/pytorch/ubi9-python-3.12/Dockerfile.konflux.rocm`
- `jupyter/rocm/tensorflow/ubi9-python-3.12/Dockerfile.konflux.rocm` *(also has a **later** site-packages `COPY` after ensure — see [#3934](https://github.com/opendatahub-io/notebooks/pull/3934) review)*

Representative duplicated steps:

```dockerfile
mkdir /opt/app-root/runtimes
rm /opt/app-root/share/jupyter/metadata/runtime-images/*.json
sed -i -e "s/Python.*/$(python --version | cut -d '.' -f-2)\",/" \
  /opt/app-root/share/jupyter/kernels/python3/kernel.json
install -D -m 0664 /opt/app-root/bin/utils/jupyter_server_config.py \
  /opt/app-root/etc/jupyter/jupyter_server_config.py
jupyter labextension disable "@jupyterlab/apputils-extension:announcements"
jupyter labextension disable jupyterlab-kubeflow-kale   # most leaves; minimal omits kale
chmod g+w /opt/app-root/etc/jupyter/labconfig
/opt/app-root/bin/utils/addons/apply.sh
/usr/local/bin/ensure-openshift-site-packages.sh
```

## Variants to parameterize

| Difference | Examples |
|------------|----------|
| `pipeline-runtimes` mkdir | datascience only |
| Kale disable | most leaves; minimal skips |
| Split vs single `RUN` | trustyai (pip in RUN 1, jupyter setup in RUN 2) |
| Post-ensure site-packages `COPY` | rocm tensorflow (`usercustomize.pth`, `monkey_patch_protobuf_6x.py`) |
| Step order | minimal reorders labextension vs sed vs install |

## Proposed direction

Pick one (or combine):

1. **`base-images/utils/setup-jupyter-workbench.sh`** — invoked once from each leaf `RUN` under `USER 1001:0` + `umask 0002`; flags/env for kale, pipeline-runtimes, etc.
2. **`scripts/dockerfile_fragments.py` marker** — extend the existing `### BEGIN` / `### END` sync mechanism ([`scripts/dockerfile_fragments.py`](scripts/dockerfile_fragments.py), documented in [`scripts/README.md`](scripts/README.md#dockerfile_fragmentspy), tested under [#2683](https://github.com/opendatahub-io/notebooks/issues/2683)) so the heredoc body is generated in all leaf Dockerfiles from one source.

Either approach should preserve the OpenShift contract documented in [#3928](https://github.com/opendatahub-io/notebooks/issues/3928) / [`ARCHITECTURE.md`](ARCHITECTURE.md): `USER 1001:0`, `umask 0002`, surgical modes at creation, `ensure-openshift-site-packages.sh` last among site-packages mutations.

## Related issues / docs

| Ref | Relevance |
|-----|-----------|
| [#3355](https://github.com/opendatahub-io/notebooks/issues/3355) | Umbrella Dockerfile dedup (Konflux twins + variant merge); complementary, not a substitute |
| [RHOAIENG-54488](https://issues.redhat.com/browse/RHOAIENG-54488) | Jira twin of #3355 ([`docs/konflux.md`](docs/konflux.md#dockerfile-deduplication)) |
| [#3928](https://github.com/opendatahub-io/notebooks/issues/3928) | Permission model this block must keep |
| [#1321](https://github.com/opendatahub-io/notebooks/issues/1321) | Replace fragile kernel `sed` with JSON-safe Python while deduplicating |
| [#2683](https://github.com/opendatahub-io/notebooks/issues/2683) | Stronger tests for `dockerfile_fragments.py` if we use markers |
| [`scripts/dockerfile_fragments.py`](scripts/dockerfile_fragments.py) | Existing codegen for PDF export deps, pip/uv config, subscription-manager, etc. |

## Acceptance criteria

- [ ] One authoritative definition of the post-pip jupyter provisioning steps
- [ ] All 10 leaf `Dockerfile.konflux.*` call it (trustyai merged back to single `RUN` or script covers both)
- [ ] OpenShift permission battery from #3928 still passes (site-packages + etc/share jupyter g+w, arbitrary UID pip)
- [ ] `#1321` kernel JSON fix can land in the shared location without touching 10 files
- [ ] `make test` / alignment checks green

## Non-goals

- Merging cpu/cuda/rocm Dockerfiles ([#3355](https://github.com/opendatahub-io/notebooks/issues/3355) phase 2) — can be done before or after this
- Changing Konflux LABEL parameterization ([#3355](https://github.com/opendatahub-io/notebooks/issues/3355) phase 1)
